src/Worker/Handler/Encapsulator/Debugger.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * This encapsulator will print to log debug information stored in registry
+ */
+declare (strict_types=1);
+
+namespace Maleficarum\Worker\Handler\Encapsulator;
+
+/**
+ * Class Debugger
+ *
+ * @package Maleficarum\Worker\Handler\Encapsulator
+ */
+class Debugger extends \Maleficarum\Worker\Handler\Encapsulator\AbstractEncapsulator {
+    /**
+     * Default time in seconds when debug information will be printed to log
+     */
+    CONST DEFAULT_TIME_AFTER_PRINT_TO_LOG = 600; //10 min
+
+    /**
+     * Default memory usage in MB when debug information will be printed to log
+     */
+    CONST DEFAULT_MEMORY_USAGE_AFTER_PRINT_TO_LOG = 80; //80MB
+
+    /**
+     * @see \Maleficarum\Worker\Handler\Encapsulator\Encapsulator::beforeHandle()
+     */
+    public function beforeHandle() : bool {
+        $registry = $this->getHandler()->getRegistry();
+        $registry['debugMessages'] = [];
+        $registry['debugInfo'] = [
+            'timeAfterPrintToLog'=> self::DEFAULT_TIME_AFTER_PRINT_TO_LOG,
+            'memoryUsageAfterPrintToLog'=> self::DEFAULT_TIME_AFTER_PRINT_TO_LOG,
+            'startTime' => \microtime(true),
+            'startMemory' => \memory_get_usage()
+        ];
+        $this->getHandler()->setRegistry($registry);
+
+        return true;
+    }
+
+    /**
+     * @see \Maleficarum\Worker\Handler\Encapsulator\Encapsulator::afterHandle()
+     */
+    public function afterHandle(bool $result): bool {
+        $registry = $this->getHandler()->getRegistry();
+        if(\count($registry['debugMessages']) === 0){
```

Comment:
This will fail if $registry does not have the 'debugMessages' index and you have no certainty that it will be there.

Reply:
Changed at 5f3f302cb54de38384d4ba0180bb30fabe25ec7e